### PR TITLE
feat: prompt server port during setup

### LINKED ISSUE #2259: API Setup should prompt for the TCP listening port
**Is your feature request related to a problem? Please describe.**

API Setup should prompt for the TCP listening port. 

**Describe the solution you'd like**

1. The API Setup should prompt for the TCP listening port and:
    1. store the value in the `.env` file
    2. make the API instance listen on this port for Admin and Mobile connections
    3. operate as designed in all situations on this TCP port
1. It must work in standalone mode or as a Docker container
1. This will allow us to run multiple instances of the API on any server
1. Tests must be created for this feature
1. When complete we'll then to evaluate possible changes to the Mobile and Admin client apps

**Describe alternatives you've considered**

- See above

**Approach to be followed (optional)**

- See above

**Additional context**

- See above

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 

- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by NamitBhutani**
**What kind of change does this PR introduce?**

Feature to prompt the user for server port , to run the talawa-api on. Works with docker environments too. Updated the docs for env.
**Issue Number:**

Fixes #2259

**Did you add tests for your changes?**

No need for test, as no new function has been written.

**Summary**

**Does this PR introduce a breaking change?**

No

**Other information**

none

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@NamitBhutani Please fix the failing tests.

**Comment by NamitBhutani** [comment]
> @NamitBhutani Please fix the failing tests.

I need to change some unauthorized files like docker-compose and setup.ts , to implement the functionality, That's why the test it failing.

**Comment by palisadoes** [comment]
1. Please fix this error:

    image

1. Please add a video showing the installation process and proof that the app runs on the new port correctly with Admin

**Comment by NamitBhutani** [comment]
[1](https://drive.google.com/file/d/1UrDMgtyUJk4pDo_byNe4fZqfw6SEepMn/view?usp=sharing)
https://drive.google.com/file/d/1IF92GtQJjhADVfoa44Tc5mOa2H8T5wfa/view?usp=sharing

**Comment by NamitBhutani** [comment]
> 1. Please fix this error:
>        image
> 
>     2. Please add a video showing the installation process and proof that the app runs on the new port correctly with Admin

The error has been fixed by adding a gh action step to set the env to 4000.

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. The red text is in the wrong location
2. The yellow text seems unnecessary
3. There is too much white space in the blue area. 

image

**Comment by palisadoes** [comment]
It otherwise looks OK

**Comment by NamitBhutani** [comment]
image
Fixed it!

> 1. The red text is in the wrong location
> 
>     2. The yellow text seems unnecessary
> 
>     3. There is too much white space in the blue area.
> 
> 
> image

**Comment by NamitBhutani** [comment]
> Second port of 4000 no need to be changed. It implies port variable of docker. 

Yes but for uniformity, it is recommended to keep docker's port same as the local port.

**Comment by NamitBhutani** [comment]
@palisadoes Is there anything else left to fix?

File: docker-compose.dev.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ services:
       context: .
       dockerfile: Dockerfile.dev
     ports:
-      - 4000:4000
+      - "${SERVER_PORT:-4000}:${SERVER_PORT:-4000}"
```
Comment (palisadoes) [inline docker-compose.dev.yaml:23]: Is `${SERVER_PORT}` the port that the external Admin instance will communicate with?
I'm assuming that Docker will automatically insert this value based on the environment variable value. Correct?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comment

File: docker-compose.dev.yaml (hunk 1 above)
Comment (NamitBhutani) [reply docker-compose.dev.yaml:23]: Yes!

**Comment by palisadoes** [comment]
Thanks. Please fix the conflicting file so that we can merge.

**Comment by NamitBhutani** [comment]
> Thanks. Please fix the conflicting file so that we can merge.

Done!

**Review by palisadoes — APPROVED** [review]

**Review by palisadoes — APPROVED** [review]

**Pull request merged**